src/Lykke.Service.ExchangeConnector/Exchanges/Concrete/Shared/OrderBooksHarvesterBase.cs

```diff
@@ -1,46 +1,78 @@
 ﻿using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Common.Log;
 using Polly;
-using TradingBot.Exchanges.Concrete.BitMEX;
+using TradingBot.Communications;
 using TradingBot.Infrastructure.Configuration;
+using TradingBot.Infrastructure.Exceptions;
 using TradingBot.Trading;
 
 namespace TradingBot.Exchanges.Concrete.Shared
 {
     internal abstract class OrderBooksHarvesterBase : IDisposable
     {
         protected readonly ILog Log;
-        protected readonly WebSocketTextMessenger Messenger;
-        protected readonly ISet<OrderBookItem> OrderBookSnapshot;
+        protected readonly OrderBookSnapshotsRepository OrderBookSnapshotsRepository;
+        protected readonly OrderBookEventsRepository OrderBookEventsRepository;
+        protected CancellationToken CancellationToken;
+
+        private readonly ConcurrentDictionary<string, OrderBookSnapshot> _orderBookSnapshots;
+        private ExchangeConverters _converters;
+        private readonly Timer _heartBeatMonitoringTimer;
+        private readonly TimeSpan _heartBeatPeriod = TimeSpan.FromSeconds(30);
```

Comment:
Is it necessary to stop and restart connection every 30 seconds? I'm afraid it'll lead to data loss between the stop and start events.

Reply:
It stops only if there were no responses|heartbeats from exchange within 30 seconds